include/interface/blas2_interface.h

```diff
@@ -142,63 +250,174 @@ typename executor_t::policy_t::event_t inline _gemv(
+ matrix, i.e.
+ computing the mathematical operation:
+
+ x = alpha*x*y + A
+
+ See the netlib blas interface documentation for more details of the high level
+ interface: http://www.netlib.org/lapack/explore-html/db/d5c/sger_8f.html
+
+ */
 template <typename executor_t, typename index_t, typename element_t,
           typename container_0_t, typename increment_t, typename container_1_t,
           typename container_2_t>
 typename executor_t::policy_t::event_t inline _ger(
-    executor_t& ex, index_t _M, index_t _N, element_t _alpha, container_0_t _vx,
-    increment_t _incx, container_1_t _vy, increment_t _incy, container_2_t _mA,
-    index_t _lda) {
+    executor_t& ex,     // executor_t (sycl, parallel, serial, etc)
+    index_t _M,         // The rows in matrix M
+    index_t _N,         // The rows of matrix N
+    element_t _alpha,   // Scalar alpha
+    container_0_t _vx,  // >(1 + (_M-1)*abs(_incx)), input vector X
+    increment_t _incx,  // Increment for vector X
+    container_1_t _vy,  // >(1 + (_N-1)*abs(_incy)), input vector Y
+    increment_t _incy,  // Increment for vector Y
+    container_2_t _mA,  // (_lda, n) array containing A, the output
+    index_t _lda        // >max(1, m), Leading dimension of A
+) {
   // TODO: Here we can use some heuristics to select localn global, local, and
   // scratch size per device
   return internal::_ger(ex, _M, _N, _alpha,
                         ex.get_policy_handler().get_buffer(_vx), _incx,
                         ex.get_policy_handler().get_buffer(_vy), _incy,
                         ex.get_policy_handler().get_buffer(_mA), _lda);
 }
+
+/*!
+ @brief Generalised vector product sum.
+
+  Generalised vector squaring followed by a sum with a rectangular symmetric
+ matrix, i.e.
+ computing the mathematical operation:
+
+ x = alpha*x*x**T + A
+
+ See the netlib blas interface documentation for more details of the high level
+ interface: http://www.netlib.org/lapack/explore-html/db/d99/ssyr2_8f.html
+
+ */
 template <typename executor_t, typename index_t, typename element_t,
           typename container_0_t, typename increment_t, typename container_1_t>
 typename executor_t::policy_t::event_t inline _syr(
-    executor_t& ex, char _Uplo, index_t _N, element_t _alpha, container_0_t _vx,
-    increment_t _incx, container_1_t _mA, index_t _lda) {
+    executor_t& ex,     // executor_t (sycl, parallel, serial, etc)
+    char _Uplo,         // Whether the matrix is upper/lower ('u', 'l')
+    index_t _N,         // >0 The order of matrix A
+    element_t _alpha,   // Scalar alpha
+    container_0_t _vx,  // (1 + (_N-1)*abs(_incx)), input vector X
+    increment_t _incx,  // !=0 The increment for the elements of X
+    container_1_t _mA,  // (_lda, _N) The output matrix
+    index_t _lda        // >max(1, _N) The first dimension of _mA
+) {
   // TODO: Here we can use some heuristics to select localn global, local, and
```

Comment:
remove this todo and instead create an internal issue

Reply:
https://jira.office.codeplay.com/browse/SYCLE-717